dll/win32/shell32/shlexec.cpp

```diff
@@ -802,6 +802,15 @@ static UINT SHELL_FindExecutable(LPCWSTR lpPath, LPCWSTR lpFile, LPCWSTR lpVerb,
         xlpFile[0] = '\0';
     }
 
+    /* Clear any trailing periods */
+    for (SIZE_T i = wcslen(xlpFile); i > 0; i--)
+    {
+        if (i > 0 && xlpFile[i - 1] == '.')
```

Comment:
```suggestion
        if (xlpFile[i - 1] == '.')
```
because when we're inside the for-loop, we know that i > 0, from the loop test.

Reply:
Since the value if ```i``` is decremented on every iteration of the for loop, my fear was that it might get to zero. In this case when we look at ```xlpFile[i - 1]``` we will be accessing ```xlpFile[-1]``` which is an array access out-of-bounds. Therefore, I prefer to keep the ```i > 0``` condition for this reason as I see it as a good defensive programming practice. You are welcome to try and convince me otherwise. I am always interested in learning. Thanks.

Reply:
You're not really defending against anything, since this is already ensured by the for-loop test.

Reply:
OK. Thanks. I do see now the the FOR loops depends upon ```i > 0``` or otherwise, it will not continue. Sorry it took me so long to understand. I have now changed the code. Thanks again.